### PR TITLE
WIP gcbmgr: add nonInteractive and tests

### PR DESCRIPTION
**Pull request by cpanato**
### What type of PR is this?
add nonInteractive variable similar have in the FF and add some tests to validate the flow

Before I move one, I would like to get some feedback on that to check if this is the right approach and if we want that.

/cc @justaugustus @saschagrunert 

/kind cleanup

#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note

```

### CODE REVIEW

**Comment by justaugustus** [comment]
@cpanato -- Needs rebase

**Comment by cpanato** [comment]
@justaugustus My idea here, before moved forward get some initial feedback if this is okay and what we want as starting point

File: cmd/krel/cmd/gcbmgr.go [hunk 1]
```diff
@@ -41,12 +41,16 @@ type gcbmgrOptions struct {
...
+
+	nonInteractive bool
 }
 
 var (
 	gcbmgrOpts = &gcbmgrOptions{}
 	buildOpts  = &build.Options{}
 
+	buildPrepareBuilds = build.PrepareBuilds
+	buildRunSingleJob  = build.RunSingleJob
```
Comment (saschagrunert) [inline cmd/krel/cmd/gcbmgr.go]: I would prefer if we move those functions into an interface and mock them via counterfeiter (see the github or git package).

File: cmd/krel/cmd/gcbmgr.go [hunk 2]
```diff
@@ -41,12 +41,16 @@ type gcbmgrOptions struct {
 	releaseType  string
 	buildVersion string
 	gcpUser      string
+
+	nonInteractive bool
```
Comment (saschagrunert) [inline cmd/krel/cmd/gcbmgr.go]: I think this is totally fine :+1:

File: cmd/krel/cmd/gcbmgr.go (hunk 1 above)
Comment (cpanato) [reply cmd/krel/cmd/gcbmgr.go]: ok, cool, i will check in how to do that :) thanks for the feedback

**Comment by cpanato** [comment]
will close that, need more time to implement

**Pull request closed without merging**